apis/Google.Cloud.Spanner.Data/docs/configuration.md

```diff
@@ -0,0 +1,180 @@
+- The session pool options, as exposed via
+  `Google.Cloud.Spanner.V1.SessionPoolOptions`.
+
+The connection string options are [specified in this
+document](connection_string.md). The remaining options relate to the
+session pool, which handles acquiring, refreshing and evicting
+sessions. Spanner sessions are relatively expensive (in terms of
+time) to create, so it's important that they are handled
+appropriately.
+
+## Session pool options explained
+
+`SessionPoolOptions` exposes the following properties:
+
+**MaximumActiveSessions**
+
+Default value: 100
+
+The maximum number of sessions that can be active per database. An
+active session is one that has been acquired but not yet released
+back to the pool. Broadly speaking, this is equivalent to the number
+of concurrent Spanner-based operations your application will be able
+to perform.
+
+Applications may wish to change this setting if they require high
+levels of concurrency.
+
+**MinimumPooledSessions**
+
+Default value: 10
+
+The minimum number of sessions to maintain in the pool of available
+sessions, on a per-database basis. If the number of pooled sessions
+falls below this number, more sessions are added automatically,
+unless that would require more than the configured maximum active
+session count.
+
+Applications may wish to change this setting if they are likely to
+encounter sudden bursts of requests, and don't wish to have to wait
+for sessions to be created.
+
+**WriteSessionsFraction**
+
+Default value: 0.2
+
+The session pool maintains sessions in two states:
+
+- Without a transaction
+- With a read/write transaction
+
+The value of this setting determines how many sessions to keep in
+each state. For example, with the default value of 0.2, roughly 20%
+of sessions will be pooled with a read/write transaction.
+
+When a session is acquired, if there is not one already available
+with the requested transaction options, a transaction can be
+requested. Using a session which was previously associated with a
+read/write transaction for operations which don't use that
+transaction does not affect the behavior; it just wastes the small
+amount of time spent creating the transaction to start with.
+
+Increasing this value will increase the average time taken to release a
+session back to the pool, as more read/write transactions will be
+acquired as part of that operation. However, acquiring sessions with
+read/write transactions from the pool will then be fast.
+
+Applications that are aware of their usage patterns ahead of time
+may wish to change this setting to reduce latency. This is
+particularly true for applications which only perform read-only
+operations (in which case this value should be set to 0) or only
+perform operations using read/write transactions (in which case this
+value should be set to 1.0).
+
+**IdleSessionRefreshDelay**
+
+Default value: 15 minutes
+
+Sessions that are unused for long periods of time can be invalidated
```

Comment:
Maybe before the whole explanation, a simple sentence like : "This is the time a session needs to be idle before it is considered for refresh."

Reply:
Done.